tests/MaintainersMetadataTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ManageApiTest;
+
+use Keboola\ManageApi\Client;
+use Keboola\ManageApi\ClientException;
+
+class MaintainersMetadataTest extends ClientTestCase
+{
+    public const TEST_METADATA = [
+        [
+            'key' => 'test_metadata_key1',
+            'value' => 'testval',
+        ],
+        [
+            'key' => 'test.metadata.key2',
+            'value' => 'testval2',
+        ],
+    ];
+
+    public const PROVIDER_USER = 'user';
+    public const PROVIDER_SYSTEM = 'system';
+
+    /**
+     * @var array
+     */
+    private $maintainer;
+
+    public function setUp()
+    {
+        /*
+         * $this->client = super-admin
+         * $this->normalUserClient = normal admin without any connections yet
+         */
+        parent::setUp();
+
+        $maintainerPrefix = sprintf('%s - MaintainerMetadata', self::TESTS_MAINTAINER_PREFIX);
+
+        // cleanup maintainers created by tests
+        foreach ($this->client->listMaintainers() as $maintainer) {
+            if (strpos($maintainer['name'], $maintainerPrefix) === 0) {
+                $this->client->deleteMaintainer($maintainer['id']);
+            }
+        }
+
+        // create default maintainer
+        $this->maintainer = $this->client->createMaintainer([
+            'name' => sprintf('%s - %s.%s', $maintainerPrefix, date('Ymd.His'), rand(1000, 9999)),
+        ]);
+    }
+
+    public function providers(): array
+    {
+        return [
+            'system provider' => [
+                self::PROVIDER_SYSTEM,
+            ],
+            'user provider' => [
+                self::PROVIDER_USER,
+            ],
+        ];
+    }
+
+    // normal user
+    public function testNormalUserCannotManageMetadata(): void
+    {
+        // User
+        $metadata = $this->createUserMetadata($this->client, $this->maintainer['id']);
+        $this->cannotManageUserMetadata($this->normalUserClient, $metadata[0]['id']);
+
+        // System
+        $this->cannotSetSystemMetadata($this->normalUserClient);
+
+        // superadmin creates system metadata
+        $metadataArray = $this->createSystemMetadata($this->client, $this->maintainer['id']);
+        // but normal user cannot delete it
+        $this->cannotDeleteMetadata($this->normalUserClient, $metadataArray[0]['id']);
+    }
+
+    // super admin
+    /**
+     * @dataProvider providers
+     */
+    public function testSuperAdminCanManageMetadata(string $provider): void
+    {
+        // Create
+        $this->createMetadata($this->client, $this->maintainer['id'], $provider);
```

Comment:
tim, ze zakladas v setup noveho maintainera pomocou `$this->client` tak sa stane automaticky maintainer admin (pretoze je pridany do toho maintianera), takze tu ked testujes ze superAdmin moze zalozit metadato v podstate zakladas superAdminom + maintainer adminom a mal by si to testovat len ako super admin.

Reply:
Ak to chces otestovat spravne tak v setup po vytvoreni maintainera musis pridat nejakeho uzivatela ktory sa vobec v testoch nepouziva aby nedoslo k chybe napr `$this->client->addUserToMaintainer($this->testMaintainerId, ['email' => 'spam+spam@keboola.com']);` musis tam niekoho pridat inac ti npojde dalsi krok odobrania superADmina

a odobrat superAdmin uzivatela `$this->client->removeUserFromOrganization($this->organization['id'], $this->superAdmin['id']);` aby bol naozaj len super admin